### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by roryhewitt**
docs: Update README.md

Completely revamp README, including ToC (generated using https://ecotrust-canada.github.io/markdown-toc/) and standardize port numbers and format.

No tests, since it's just documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,24 +6,101 @@
...
+## Overview
+
+The Broker consists of two parts
+
+- **Broker Client**: A Docker image deployed in your infrastructure.
+- **Broker Server**: An application deployed in Snyks infrastructure.
+
+(as the Broker Server is provided by Snyk and no installation is required on your part, all further descriptions in this README refer only to the Broker Client unless stated otherwise)
+
+The Broker Client establishes a websocket 'tunnel' to the Broker Server, proxying requests from snyk.io to your SCM (fetching manifest files from monitored repositories), and vice versa (webhooks posted by the SCM). Because this tunnel is established by the Broker Client making an outbound request, there is no need to grant permissions to specific Snyk IP addresses in your firewall.
```
Comment (ipsi) [inline README.md:73]: It's not strictly for SCMs - For example, Artifactory, Nexus, and Jira aren't SCM systems. I'm not sure what the preferred term would be instead though - possibly "backend system"?

**Review by ipsi — DISMISSED** [review]
Changes look good - I really appreciate the effort you've put into this! One comment about terminology, but overall looks good.

File: README.md (hunk 1 above)
Comment (roryhewitt) [reply README.md:73]: OK, I'll update and do another PR.

Probably change the top section to reference "SCM or other backend system, such as Jira or Artifactory" and then use "backend system" throughout the rest of the doc.

BTW, any issues with leaving the small "ToC created by {third party}" link in there? It's auto-added by the ToC generator, and I can remove it if Snyk policy is to not reference stuff like that... At the same point, I *did* use that ToC generator, so...

**Review by ipsi — DISMISSED** [review]

**Comment by Jimimaku** [comment]
> [CLA assistant check](https://cla-assistant.io/snyk/broker?pullRequest=442) All committers have signed the CLA.

**Comment by Jimimaku** [comment]
> Changes look good - I really appreciate the effort you've put into this! One comment about terminology, but overall looks good.

**Comment by Jimimaku** [comment]
> docs: Update README.md
> 
> Completely revamp README, including ToC (generated using https://ecotrust-canada.github.io/markdown-toc/) and standardize port numbers and format.
> 
> No tests, since it's just documentation.

**Review by aarlaud — APPROVED** [review]

**Comment by aarlaud** [comment]
closing this as the docs moved to snyk gitbook. please drop a note if anything.

**Pull request closed without merging**